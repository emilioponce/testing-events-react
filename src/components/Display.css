.Positive,
.Negative {
  font-size: 200px;
}

.Negative {
  color: red;
}
