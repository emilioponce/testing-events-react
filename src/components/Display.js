import React from "react";

import "./Display.css";

const Display = props => (
  <div>
    <div className={props.value >= 0 ? "Positive" : "Negative"}>
      {props.value}
    </div>
  </div>
);

export default Display;
