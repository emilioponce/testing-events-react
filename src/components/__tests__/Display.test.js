import React from "react";
import ReactDOM from "react-dom";
import Display from "../Display";
import renderer from "react-test-renderer";

const component = <Display value="1" />;

describe("Component Display", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("snapshot renders correctly", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Values below 0 change its color", () => {
    const testRenderer = renderer.create(<Display value="-1" />);
    const testInstance = testRenderer.root;
    expect(testInstance.findByProps({ className: "Negative" })).toBeDefined();
  });
});
