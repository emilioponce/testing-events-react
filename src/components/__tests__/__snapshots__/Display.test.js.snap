// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Display snapshot renders correctly 1`] = `
<div>
  <div
    className="Positive"
  >
    1
  </div>
</div>
`;
